yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -549,6 +549,21 @@ private[spark] class Client(
       appMasterOnly = true)
     require(confLocalizedPath != null)
 
+    // Uploading $SPARK_CONF_DIR/log4j.properties file to the distributed cache to make sure that
+    // the executors will use the latest configurations instead of the default values. This is
+    // required when user changes log4j.properties directly to set the log configurations. If
+    // configuration file is provided through --files then executors will be taking configurations
+    // from --files instead of $SPARK_CONF_DIR/log4j.properties.
+
+    // Also uploading metrics.properties to distributed cache if exists in classpath.
+    // If user specify this file using --files then executors will use the one
+    // from --files instead.
+    for { prop <- Seq("log4j.properties", "metrics.properties")
```

Comment:
Hi @tgravescs , I just did a quick test on this. 

If oldLog4jConf points to the same log4j file under `<SPARK_HOME>/conf`, it will be added to distributed cache once and get a warning for the following one. If oldLog4jConf points to a different log4j file other than the default one under `<SPARK_HOME>/conf`, so the one under `conf` took precedence. 

I think since SPARK_LOG4J_CONF is deprecated, so there should be no problem, and semantically still keep the consistent.